src/job_metadata.c

```diff
@@ -642,6 +642,7 @@ cron_unschedule(PG_FUNCTION_ARGS)
 	cronSchemaId = get_namespace_oid(CRON_SCHEMA_NAME, false);
 	cronJobIndexId = get_relname_relid(JOB_ID_INDEX_NAME, cronSchemaId);
 
+	CachedCronJobRelationId = InvalidOid;
```

Comment:
[marcoslot](https://github.com/marcoslot) I think a better approach will be to not cache the value for these functions. What do you think?

Reply:
possibly, though I think the main reason we cache it is to make InvalidateJobCacheCallback work

Reply:
probably fair to say that caching within the schedule/unschedule functions is completely unnecessary, so maybe we should have a not-cached version of CronJobRelationId

Reply:
Changed to only caching the oid if EnableCronJobRelationIdCaching is set. This flag is set only when InvalidateJobCacheCallback is registered.